app/assets/javascripts/pages/data_portal/DataPortalReportPage.js

```diff
@@ -0,0 +1,153 @@
+(function (App) {
+  App.Page.DataPortalReportPage = Backbone.View.extend({
+
+    el: 'body',
+
+    defaults: {
+      indicators: []
+    },
+
+    initialize: function (settings) {
+      this.options = _.extend({}, this.defaults, settings);
+
+      this._setVars();
+      this._fetchData();
+    },
+
+    _setVars: function () {
+      this.indicatorsCollection = new App.Collection.IndicatorsCollection();
+
+      this.widgetsContainer = this.el.querySelector('.js-widgets');
+    },
+
+    /**
+     * Fetch the model for the selected country and year
+     */
+    _fetchData: function () {
+      // The first time we render to have the placeholder flags
+      this.render();
+
+      $.when.apply($,
+        this.indicatorsCollection.fetch()
+      ).done(function () {
+          this._loadingError = false;
+        }.bind(this))
+        .fail(function () {
+          this._loadingError = true;
+        }.bind(this))
+        .always(this.render.bind(this));
+    },
+
+    /**
+     * Return the sorted list of visible indicators
+     * NOTE: the access indicators are always displayed first
+     * @returns {object[]} widgets
+     */
+    _getVisibleIndicators: function () {
+      return this.indicatorsCollection.toJSON().filter(function (indicator) {
+        return indicator.visible;
+      }).sort(function (a, b) {
+        var aIsAccess = a.category === App.Helper.Indicators.CATEGORIES.ACCESS;
+        var bIsAccess = b.category === App.Helper.Indicators.CATEGORIES.ACCESS;
+        if (aIsAccess && !bIsAccess) return -1;
+        if (!aIsAccess && bIsAccess) return 1;
+        return 0;
+      });
+    },
+
+    /**
+     * Create the widget containers
+     * @param {number} count - number of containers to create
+     * @returns {DocumentFragment} containers
+     */
+    _createWidgetsContainer: function (count) {
+      var fragment = new DocumentFragment();
+
+      for (var i = 0, j = count; i < j; i++) {
+        var div = document.createElement('div');
+        div.classList.add('grid-s-12', 'grid-l-6');
+        var widget = document.createElement('div');
+        widget.classList.add('c-widget');
+        div.appendChild(widget);
+        fragment.appendChild(div);
+      }
+
+      return fragment;
+    },
+
+    /**
+     * Update the size of the widget container depending on
+     * if the category of the indicator
+     * @param {object} indicator
+     * @param {object} widgetContainer
+     */
+    _updateWidgetContainer: function (indicator, widgetContainer) {
+      var visibleIndicators = this._getVisibleIndicators();
```

Comment:
This won't work. If one of the report's widget is an indicator which is hidden by default, you won't be able to find it, and its container might have the wrong dimension. Instead, just use the full list of indicators: `this.indicatorsCollection.toJSON()`.

It also means you don't need the method `_getVisibleIndicators` anymore.